sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -976,13 +976,36 @@ private async Task<AmqpSession> CreateAndOpenSessionAsync(AmqpConnection connect
         ///
         /// <param name="target">The target AMQP object to open.</param>
         /// <param name="timeout">The timeout to apply when opening the link.</param>
+        /// <param name="cancellationToken">Token to signal cancellation of the operation.</param>
         protected virtual async Task OpenAmqpObjectAsync(
```

Comment:
Interesting.   Is this because of the session behavior in Service Bus, or were you seeing link creation hang in more general cases?

Reply:
Both session and non-session call into here, but the cancellationToken is only settable by users for the session case. For non-session, even it was exposed to users, it would be dangerous because we attempt to reconnect the link in case of the link being disconnected, so it would probably not make sense to use the CT in those cases.